keras_retinanet/utils/eval.py

```diff
@@ -54,7 +90,7 @@ def _compute_ap(recall, precision):
     return ap
 
 
-def _get_detections(generator, model, score_threshold=0.05, max_detections=100, save_path=None):
+def _get_detections(generator, model, score_threshold=0.05, max_detections=100, save_path=None, writer=None, steps=0, number=0):
```

Comment:
Perhaps it's better to add an argument functor, so that you can execute any function on a predicted image. That way we are hiding the `tf` specific code from this generic function.